resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -216,8 +216,70 @@ private[spark] class Client(
                 .endContainer()
               .endSpec()
             .done()
-          submitCompletedFuture.get(30, TimeUnit.SECONDS)
-        }
+          try {
+            submitCompletedFuture.get(30, TimeUnit.SECONDS)
```

Comment:
Hmm. I guess we do need to time-bound the starting up of the driver, because the client side needs to submit files to it. I'm just wondering how we expect queueing of jobs to work, when a cluster is overloaded. It seems like a user can only submit jobs if his driver is guaranteed to start running almost immediately.